boa/src/syntax/lexer/token.rs

```diff
@@ -247,6 +247,9 @@ pub enum TokenKind {
 
     /// Indicates the end of a line (`\n`).
     LineTerminator,
+
+    /// Indicates a comment, the content isn't stored.
+    Comment,
```

Comment:
I don't think we need comments as tokens. A comment can only generate a line terminator token.

Reply:
I needed something to return from the comment lex impl and this seemed to be the cleanest way.

Reply:
Hmm we could just return `()`, right?

Reply:
> Hmm we could just return `()`, right?

It expects a token because of the trait signature

Reply:
Hmmm actually, all comments return a line terminator, right?
Well, a single line comment, in theory won't be consuming the rightmost new line, but we can assume there will be one (a new line at the end of a file doesn't matter much):

https://tc39.es/ecma262/#sec-comments

Reply:
Do we return line terminators? I thought we kinda just ignored them

Reply:
What about the case https://jsfiddle.net/op7t13dc/